crates/ruff_benchmark/benches/red_knot.rs

```diff
@@ -76,12 +147,18 @@ fn setup_case() -> Case {
 
     let re_path = tomllib_path("_re.py");
     let re = system_path_to_file(&db, &re_path).unwrap();
-    Case {
+    let case = Case {
         db,
         fs,
         re,
         re_path,
-    }
+    };
+    // Run check once to assert the results. In the actual benchmark,
+    // we just check that the number of diagnostics reported is what we
+    // expect, to avoid doing most costly validation.
+    let result: Vec<_> = case.db.check().unwrap();
+    assert_diagnostics(&case.db, result);
```

Comment:
On second thought, I think the bogus codspeed numbers are telling us why we can't make this change. It turns the "cold" benchmark into a "no-op incremental" benchmark, which, not surprisingly, is very fast indeed. I don't see an easy way to fix this; that's probably why the full diagnostics assertion was included in the benchmark. I think we either need to put it back the way it was, or do it as a totally separate prior step, not part of the per-iteration `setup_case` at all.

Reply:
Aye. I'll switch it back. I guess it should be fine now since I'm not using `insta` any more. Although I guess the diagnostic strings are now longer, so comparisons will be longer.

Reply:
If we go with key fields instead of the full string, it should be pretty similar to before?

Reply:
Though doing it as a separate step (not in `setup_case`) also seems pretty easy and fine.

Reply:
Yeah sorry I made that comment before the other one.